File: 5KYU/Product of consecutive Fib numbers/solution.py
def productFib(n):
    a = 0
    b = 1
    while True:
        if n == (a * b):
            return [a, b, True]
        elif (a * b) > n:
            return [a, b, False]
        else:
            t = a
            a = b
            b = t +a

File: 5KYU/Product of consecutive Fib numbers/solution.go
package kata

func ProductFib(prod uint64) [3]uint64 {
  a := uint64 (0)
  b := uint64(1)
  
  for {
    if prod == a * b {
      return [3]uint64{a, b, 1}
    } else if a * b > prod {
      return [3]uint64{a, b, 0}
    } else {
      t := a
      a = b
      b = t + a
    }
  }
  return [3]uint64{}
}